Formula/qt6.rb

```diff
@@ -0,0 +1,172 @@
+
+    # Some config scripts will only find Qt in a "Frameworks" folder
+    frameworks.install_symlink Dir["#{lib}/*.framework"]
+
+    inreplace lib/"cmake/Qt6/qt.toolchain.cmake", /.*set.__qt_initial_.*/, ""
+
+    # The pkg-config files installed suggest that headers can be found in the
+    # `include` directory. Make this so by creating symlinks from `include` to
+    # the Frameworks' Headers folders.
+    Pathname.glob("#{lib}/*.framework/Headers") do |path|
+      include.install_symlink path => path.parent.basename(".framework")
+    end
+
+    # FIXME: this step just moves `*.app` bundles into `libexec` and create link
+    libexec.mkpath
+    Pathname.glob("#{bin}/*.app") do |app|
+      mv app, libexec
+      bin.install_symlink "#{libexec/app.stem}.app/Contents/MacOS/#{app.stem}"
+    end
+  end
+
+  test do
+    assert_equal HOMEBREW_PREFIX, shell_output("qmake -query QT_INSTALL_PREFIX").strip
+
+    (testpath/"CMakeLists.txt").write <<~EOS
+      cmake_minimum_required(VERSION 3.16.0)
+
+      project(test VERSION 1.0.0 LANGUAGES CXX)
+
+      set(CMAKE_CXX_STANDARD 17)
+      set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+      set(CMAKE_AUTOMOC ON)
+      set(CMAKE_AUTORCC ON)
+      set(CMAKE_AUTOUIC ON)
+
+      find_package(Qt6 COMPONENTS Widgets REQUIRED)
+
+      add_executable(test
+          main.cpp
+      )
+
+      target_link_libraries(test PRIVATE Qt6::Widgets)
+    EOS
+
+    (testpath/"test.pro").write <<~EOS
+      QT       += core
+      QT       -= gui
+      TARGET = test
+      CONFIG   += console
+      CONFIG   -= app_bundle
+      TEMPLATE = app
+      SOURCES += main.cpp
+    EOS
+
+    (testpath/"main.cpp").write <<~EOS
+      #include <QCoreApplication>
+      #include <QDebug>
+
+      int main(int argc, char *argv[])
+      {
+        QCoreApplication a(argc, argv);
+        qDebug() << "Hello World!";
+        return 0;
+      }
+    EOS
+
+    system "cmake", testpath
+    system "make"
+    assert_equal "Hello World!", shell_output("#{testpath}/test 2>&1").strip
+
+    # Work around "error: no member named 'signbit' in the global namespace"
+    ENV.delete "CPATH"
+    system bin/"qmake", testpath/"test.pro"
+    system "make"
+    assert_equal "Hello World!", shell_output("#{testpath}/test 2>&1").strip
+  end
+end
+
+__END__
+diff --git a/qtbase/src/corelib/global/qlibraryinfo.cpp b/qtbase/src/corelib/global/qlibraryinfo.cpp
+index bb1eafe..e433ac4 100644
+--- a/qtbase/src/corelib/global/qlibraryinfo.cpp
++++ b/qtbase/src/corelib/global/qlibraryinfo.cpp
+@@ -600,7 +600,7 @@ QString qmake_abslocation();
+ 
+ static QString getPrefixFromHostBinDir(const char *hostBinDirToPrefixPath)
+ {
+-    const QString canonicalQMakePath = QFileInfo(qmake_abslocation()).canonicalPath();
++    const QString canonicalQMakePath = QFileInfo(qmake_abslocation()).absolutePath();
+     return QDir::cleanPath(canonicalQMakePath + QLatin1Char('/')
+                            + QLatin1String(hostBinDirToPrefixPath));
+ }
```

Comment:
Has this been submitted upstream?

Reply:
I think this patch will not accept by upstream, it modify the behavior of qmake, in order to comply with homebrew's rule

Reply:
We'll have to try anyway, homebrew policy requires all patches to be submitted upstream.

Reply:
This patch can be avoided, by create a hardlink /usr/local/bin/qmake -> /usr/local/Cellar/qt/6.0.1/bin/qmake. But build environment has not permission to create hardlink in /usr/local/bin. Qt default assume user has write permission in prefix, that also why in this formula use an extra argument `--extprefix`